pkg/istio-agent/sds-agent.go

```diff
@@ -261,6 +261,74 @@ func gatewaySdsExists() bool {
 	return !os.IsNotExist(err)
 }
 
+// explicit code to determine the root CA to be configured in bootstrap file.
+// It may be different from the CA for the cert server - which is based on CA_ADDR
+// Replaces logic in the template:
+//                 {{- if .provisioned_cert }}
+//                  "filename": "{{(printf "%s%s" .provisioned_cert "/root-cert.pem") }}"
+//                  {{- else if eq .pilot_cert_provider "kubernetes" }}
+//                  "filename": "./var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+//                  {{- else if eq .pilot_cert_provider "istiod" }}
+//                  "filename": "./var/run/secrets/istio/root-cert.pem"
+//                  {{- end }}
+//
+// In addition it deals with the case the XDS server is on port 443, expected with a proper cert.
+// /etc/ssl/certs/ca-certificates.crt
+//
+// TODO: additional checks for existence. Fail early, instead of obscure envoy errors.
+func (sa *Agent) FindRootCAForXDS() string {
+	if strings.HasSuffix(sa.proxyConfig.DiscoveryAddress, ":443") {
+		return "/etc/ssl/certs/ca-certificates.crt"
+	} else if _, err := os.Stat("./etc/certs/root-cert.pem"); err == nil {
```

Comment:
Fixed to make it explicit ( and allow XDS root to be configured exactly, without guessing )

Reply:
I still have concerns about this. In what situation do we want to use this file for the XDS root cert? I feel like this is wrong in all cases - or maybe I just don't know the case where it is right. But I am pretty sure in the standard case of just setting `.values.global.mountMtlsCerts` (mount cert to /etc/certs) this will break

Reply:
Making it explicit doesn't really help if we don't actually set the explicit option

Reply:
This will be set explicitly by injector or startup script (VM) - if the XDS address is using a custom file for the root. It doesn't have to be /etc/certs. Right now we don't have a way to specify 'use this specific file' - we just use our own paths. 

Of course - the main use case will be to set it to the location of the platform credentials.

Reply:
In the current PR, if I set `.values.global.mountMtlsCerts` I am pretty sure things will break. Because we will use the mounted certs which are not the XDS root cert. Maybe in some future it will work but today, AFAIK, it will break?

Reply:
I don't think it will break - if XDS_ROOT is not set, /etc/certs is the second place we look for.

Reply:
And XDS_ROOT is not set by us, so that first case essentially doesn't matter.